csharp/src/Apache.Arrow/Field.Builder.cs

```diff
@@ -52,6 +55,131 @@ public Builder DataType(IArrowType type)
                 return this;
             }
 
+            // Build DataType from C# Type
+            public Builder DataType(System.Type valueType, string timezone = null)
+            {
+                System.Type[] genericArgs;
+                System.Type child = System.Nullable.GetUnderlyingType(valueType);
+                Type elementType;
+
+                if (child != null)
+                {
+                    // Nullable Type
+                    DataType(child, timezone);
+                    Nullable(true);
+                    return this;
+                }
+
+                // Not Nullable Type
+                Nullable(false);
+
+                switch (valueType)
+                {
+                    // Binary
+                    case var _ when valueType == typeof(byte):
+                        DataType(new BinaryType());
+                        break;
+                    case var _ when valueType == typeof(IEnumerable<byte>):
```

Comment:
You used `IEnumerable<byte>` here but `byte[]` above in the `BuildArray` method.

Reply:
yeah indeed, fixed